tls/s2n_server_finished.c

```diff
@@ -26,58 +26,66 @@
 
 #include "utils/s2n_safety.h"
 
-int s2n_server_finished_recv(struct s2n_connection *conn)
+S2N_RESULT s2n_finished_recv(struct s2n_connection *conn, uint8_t *local_verify_data)
 {
-    uint8_t *our_version;
-    int length = S2N_TLS_FINISHED_LEN;
-    our_version = conn->handshake.server_finished;
+    RESULT_ENSURE_REF(conn);
 
+    uint8_t length = conn->handshake.finished_len;
```

Comment:
since its recalculated below. also should we be calling s2n_handshake_set_finished_len?

```suggestion
    uint8_t length = 0;
```

Reply:
It's not being reset here, it's just being checked against the known correct value. I removed the "length =" line, the diff is just awful.

Reply:
oh my bad. its being checked not set.